### PR TITLE
docs(server): add example for early spiking middleware

### PR DESCRIPTION
**Pull request by manniL**
### Linked issue

#3561 

### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

This PR adds an example that showcases the usage of `next` inside a server middleware.

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by pi0** [comment]
With h3, also not calling next() without end() or returning would have same effect. Any reason to advice using legacy callback syntax?

**Comment by manniL** [comment]
> With h3, also not calling next() without end() or returning would have same effect. Any reason to advice using legacy callback syntax?

Not at all, just the lack of knowledge 😄 
So just returning (or calling `end()`) would be better?

**Comment by pi0** [comment]
Yeah you could simply use `return;` :D I'm planning to also move from `(req, res)` to `(event)` syntax progressively so would be nice if we avoid advising calbacks like next when possible

**Comment by manniL** [comment]
> Yeah you could simply use `return;` :D I'm planning to also move from `(req, res)` to `(event)` syntax progressively so would be nice if we avoid advising calbacks like next when possible

Good to know! Will update the docs then accordingly.

File: docs/content/3.docs/2.directory-structure/13.server.md [hunk 1]
```diff
@@ -89,4 +89,17 @@ export default async (req: IncomingMessage, res: ServerResponse) => {
 }
 ```
 
+To pass the request deeper into the application, you can simply `return` inside the function: 
+
+```js
+export default async (req, res, next) => {
```
Comment (pi0) [inline docs/content/3.docs/2.directory-structure/13.server.md:95]:
```suggestion
export default async (req, res) => {
```

**Comment by pi0** [comment]
Thanks!

**Pull request merged**